vpt/sinks/sqlite_store/__init__.py

```diff
@@ -0,0 +1,91 @@
+import sqlite3
+import time
+from typing import Union
+
+import keyboard
+import mouse
+
+from vpt.sinks.base import SinkBase
+from vpt.sources.base import SourceBase
+
+
+class SQLiteStore(SinkBase):
+    '''Persistently store the engagement level and keyboard/mouse events
+       in an SQLite database.'''
+    connection: sqlite3.Connection
+
+    def __init__(self, db_path: str, mouse_source: SourceBase,
+                 keyboard_source: SourceBase, engagement_source: SourceBase):
+        '''Create a database or open an existing one.'''
+        self.connection = sqlite3.connect(db_path)
+        c = self.connection.cursor()
+        c.execute('''
+            CREATE TABLE IF NOT EXISTS engagement (
+                code integer NOT NULL,
+                timestamp integer NOT NULL
+            )
+        ''')
+        c.execute('''
+            CREATE TABLE IF NOT EXISTS keystrokes (
+                keycode integer NOT NULL,
+                modifiers text NOT NULL,
+                timestamp integer NOT NULL
+            )
+        ''')
+        c.execute('''
+            CREATE TABLE IF NOT EXISTS mouse_events (
+                type text NOT NULL,
+                x real,
```

Comment:
By the way, why are we storing the position of the mouse in floating-point values? I think it should be `int` since stuff on the screen can be only located on discrete pixels. `math.round` the mouse coordinates in the `MouseSource` and the number of events drops down significantly.